### PR TITLE
Up for review

### PR DESCRIPTION
**Pull request by Arti-Khillare1**

### CODE REVIEW

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.html [hunk 1]
```diff
@@ -1 +1,153 @@
...
+      <div color="secondary" class="fixed top-32 w-full z-50 border-b px-10 py-2 text-lg text-tertiary bg-white">
+        <span>{{ "CERTIFICATE" | translate }}</span>
+      </div>
+      <div class="px-10 mt-20 mb-4 text-lg text-tertiary">
+        <p>
+          <span class="required">{{ "CERTIFICATE_DESCRIPTION" | translate }}</span>
+          <span class="px-1 text-base text-[$buttonDisable]">({{ "CERTIFICATE_SUB_DESCRIPTION" | translate }})</span>
+        </p>
+        <mat-radio-group color="primary" aria-label="Select an option" formControlName="selectedOption">
+          <mat-radio-button value="1" class="-mx-1 mr-6">{{ 'YES' | translate }}</mat-radio-button>
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.html]: please give it in for loop

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.html [hunk 2]
```diff
@@ -1 +1,153 @@
...
+          <mat-radio-button value="2" class="-mx-1">{{ 'No' | translate }}</mat-radio-button>
+        </mat-radio-group>
+      </div>
+  
+      @if(certificateForm.get('selectedOption')?.value === '1') {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.html]: why "1" string comparision ?

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.disabled-class {
+    cursor:not-allowed;
+    pointer-events: none;
+    opacity: 0.5; 
+    color:$buttonDisable;
+}
+
+.error-message {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss:19]: isn't this available in styles.scss file ?

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    font-size: 12px;
+    padding: 0.125rem 0;
+}
+
+.error-icon {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss:27]: same here ?

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts [hunk 5]
```diff
@@ -1,12 +1,134 @@
-import { Component } from '@angular/core';
-
+import { Component, OnInit } from '@angular/core';
+import { TranslateModule } from '@ngx-translate/core';
+import { MatIconModule } from '@angular/material/icon';
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts:3]: import all this as once .

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts [hunk 6]
```diff
@@ -1,12 +1,134 @@
...
+  constructor(private dialog : MatDialog, private fb: FormBuilder, private formService: FormService,){}
+
+  ngOnInit() {
+    this.initForm();
+    this.getCertificateForm();
+  }
+
+  initForm() {
+    this.certificateForm = this.fb.group({
+      selectedOption: [''],
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts:38]: why the hard coded form here ?

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts [hunk 7]
```diff
@@ -1,12 +1,134 @@
...
+
+  removeLogo(index: number) {
+    this.attachLogo.splice(index, 1);
+  }
+
+  removeSignature(index: number) {
+    this.attachSign.splice(index, 1);
+  }
+
+  get attachedLogosData(): FormArray {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts:96]: keep as one function and pass FormControl as parameter and return the the data as FormArray

File: projects/lib-shared-modules/src/lib/components/dialogs/dialog-popup/dialog-popup.component.html [hunk 8]
```diff
@@ -51,6 +51,64 @@ <h3 class="text-sm font-bold p-4">{{dialogueData?.content |translate}}</h3>
...
+                (change)="onFileSelected($event)"
+                class="hidden"
+                #fileInput
+              />
+              <button  type="button" class="flex justify-center items-center button border rounded bg-lightGreen text-sm text-white w-48 min-w-[192px] p-2"
+              (click)="fileInput.click()">
+                <mat-icon class="cursor-pointer text-sm">vertical_align_top</mat-icon>
+                <span>{{"UPLOAD_BUTTON" | translate}}</span>
+              </button>
+              <p class="text-xs flex items-center justify-center pt-1">upload PNG upto 50kb</p>
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/dialogs/dialog-popup/dialog-popup.component.html]: please add translate here

File: projects/lib-shared-modules/src/lib/services/form/form.service.ts [hunk 9]
```diff
@@ -108,6 +108,23 @@ export class FormService {
     }
   }
 
+  getCertificateForm(){
+    return new Promise((resolve, reject) => {
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/services/form/form.service.ts:112]: why return as promise ? 
why this function first of all ?
there is a common function to use to get forms use that.

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss (hunk 3 above)
Comment (Arti-Khillare1) [reply projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss:19]: this is not available in style.scss

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss (hunk 4 above)
Comment (Arti-Khillare1) [reply projects/lib-project/src/lib/dock/components/certificates/certificates.component.scss:27]: this is also not in style.scss

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**